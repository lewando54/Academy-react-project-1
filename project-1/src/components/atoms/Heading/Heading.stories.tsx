/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import Heading from './Heading'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: Heading,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    level: { control: 'select', options: [1, 2, 3, 4, 5, 6] },
    children: { control: 'text' }
  }
} as ComponentMeta<typeof Heading>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Heading> = (args) => <Heading {...args} />

export const Primary = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
