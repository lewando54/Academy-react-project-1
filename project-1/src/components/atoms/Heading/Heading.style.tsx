import styled from 'styled-components'

// Heading tylko, nie title

const StyledHeading = styled.div`
    width: 100%;
    text-transform: uppercase;
    text-align: left;
    font-weight: 700;
    font-size: 1rem;
`

export default StyledHeading
