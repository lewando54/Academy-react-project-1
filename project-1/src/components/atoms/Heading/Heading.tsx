import React from 'react'
import StyledHeading from './Heading.style'

export interface IHeadingProps {
  level?: 1 | 2 | 3 | 4 | 5 | 6
  children?: React.ReactNode | string
  testId: string
}

function Heading ({ level = 1, children, testId }: IHeadingProps): React.ReactElement {
  return (
    <StyledHeading as={`h${level}`} data-testid={testId}>
      {children}
    </StyledHeading>
  )
}

export default Heading
