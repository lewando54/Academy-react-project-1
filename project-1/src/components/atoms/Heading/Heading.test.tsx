import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Heading from './Heading'

describe('Heading', () => {
  it('renders correctly', () => {
    const component = render(<Heading>Testing</Heading>)
    expect(component.getByText('Testing')).toBeInTheDocument()
  })

  it('renders correct html heading tag', () => {
    /* const levels: Array<IHeadingProps['level']> = [1, 2, 3, 4, 5, 6]
    for (let i = 0; i < levels.length; i++) {
      const component = render(<Heading level={levels[i]}>Testing</Heading>)
      expect(component.container.querySelector(`h${i}`)).toBeInTheDocument()
    } */
    const component = render(<Heading level={1}>Testing</Heading>)
    expect(component.container.querySelector('h1')).toBeInTheDocument()
    const component2 = render(<Heading level={2}>Testing</Heading>)
    expect(component2.container.querySelector('h2')).toBeInTheDocument()
    const component3 = render(<Heading level={3}>Testing</Heading>)
    expect(component3.container.querySelector('h3')).toBeInTheDocument()
    const component4 = render(<Heading level={4}>Testing</Heading>)
    expect(component4.container.querySelector('h4')).toBeInTheDocument()
    const component5 = render(<Heading level={5}>Testing</Heading>)
    expect(component5.container.querySelector('h5')).toBeInTheDocument()
    const component6 = render(<Heading level={6}>Testing</Heading>)
    expect(component6.container.querySelector('h6')).toBeInTheDocument()
  })
})
