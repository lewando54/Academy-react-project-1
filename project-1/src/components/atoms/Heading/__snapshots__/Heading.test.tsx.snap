// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render <h1> tag 1`] = `
<h1
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h1>
`;

exports[`Heading should render <h1> tag by default 1`] = `
<h1
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h1>
`;

exports[`Heading should render <h2> tag 1`] = `
<h2
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h2>
`;

exports[`Heading should render <h3> tag 1`] = `
<h3
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h3>
`;

exports[`Heading should render <h4> tag 1`] = `
<h4
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h4>
`;

exports[`Heading should render <h5> tag 1`] = `
<h5
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h5>
`;

exports[`Heading should render <h6> tag 1`] = `
<h6
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h6>
`;

exports[`Heading should render correctly 1`] = `
<h1
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h1>
`;

exports[`Heading should render innerHTML 1`] = `
<h1
  className="sc-bcXHqe klWJc"
  data-testid="heading"
>
  Testing
</h1>
`;
